web/packages/teleterm/src/ui/components/OfflineGateway.tsx

```diff
@@ -21,71 +21,77 @@ import { ButtonPrimary, Flex, Text } from 'design';
 
 import * as Alerts from 'design/Alert';
 import Validation from 'shared/components/Validation';
-
-import LinearProgress from 'teleterm/ui/components/LinearProgress';
-
-import { DocumentGatewayProps } from '../DocumentGateway/DocumentGateway';
+import { Attempt } from 'shared/hooks/useAsync';
 
 import { PortFieldInput } from './FieldInputs';
 
-type OfflineDocumentGatewayProps = Pick<
-  DocumentGatewayProps,
-  'connectAttempt' | 'defaultPort' | 'reconnect'
->;
+export function OfflineGateway(props: {
+  connectAttempt: Attempt<void>;
+  /** Setting `isSupported` to false hides the port input. */
+  gatewayPort:
+    | { isSupported: true; defaultPort: string }
+    | { isSupported: false };
+  reconnect(port?: string): void;
+  /** Gateway target displayed in the UI, for example, 'cockroachdb'. */
+  targetName: string;
+  /** Gateway kind displayed in the UI, for example, 'database'. */
+  gatewayKind: string;
+}) {
+  const defaultPort = props.gatewayPort.isSupported
+    ? props.gatewayPort.defaultPort
+    : undefined;
+
+  const [port, setPort] = useState(defaultPort);
+  const [reconnectRequested, setReconnectRequested] = useState(false);
 
-export function OfflineGateway(props: OfflineDocumentGatewayProps) {
-  const [port, setPort] = useState(props.defaultPort);
-  const statusDescription =
-    props.connectAttempt.status === 'processing' ? 'being set up' : 'offline';
   const isProcessing = props.connectAttempt.status === 'processing';
-  const shouldShowPortInput =
-    props.connectAttempt.status === 'error' || port !== props.defaultPort;
+  const statusDescription = isProcessing ? 'being set up…' : 'offline.';
+  const shouldShowReconnectControls =
```

Comment:
I decided to change that because now I can also hide the Reconnect button if the request is pending and the user hasn't clicked it yet.

Reply:
Good change!